app/src/main/java/org/dieschnittstelle/mobile/android/skeleton/MainActivity.java

```diff
@@ -1,34 +1,93 @@
+    private User user = new User("", "");
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        viewModel = new ViewModelProvider(this).get(MainViewModel.class);
+
+        ActivityMainBinding MainViewBinding = DataBindingUtil.setContentView(
+                this,
+                R.layout.activity_main
+        );
+        MainViewBinding.setMainViewModel(this.viewModel);
+        MainViewBinding.setLifecycleOwner(this);
+
+        userDBOperation = ((TaskApplication) getApplication()).getUserDatabaseOperation();
+        viewModel.setUserDBOperation(userDBOperation);
+
+        viewModel.setUser(user);
+
+        userDbInitialise(); // insert testdata
 
         welcomeText = findViewById(R.id.welcomeText);
         welcomeText.setText(R.string.welcome_message);
-        welcomeText.setOnClickListener(view -> this.showTaskListView());
 
-        showTaskListAction = findViewById(R.id.showTaskListAction);
-        showTaskListAction.setOnClickListener(view -> this.showTaskListView());
+        emailAdress = findViewById(R.id.editTextTextEmailAddress);
+        password = findViewById(R.id.editTextTextPassword);
+
+        // Login
+        this.viewModel.getLoginLiveData().observe(this, onLogin -> {
+            if (onLogin == "usernameIsNotAnEMail") {
+                showPersistentMessage(getString(R.string.login_username_error));
+            } else if (onLogin == "userAuthenticationIsFailed"){
+                showPersistentMessage(getString(R.string.login_authentication_failed));
+            } else if (onLogin == "passwordIsNotAllowed"){
+                showPersistentMessage(getString(R.string.login_password_not_allowed));
+            } else if (onLogin == "userIsAuthenticated") {
+                Intent callTaskOverviewIntent = new Intent(this, TaskListViewActivity.class);
+                startActivity(callTaskOverviewIntent);
+                this.finish();
+            }
+        });
     }
 
-    protected void showTaskListView() {
-        Intent callTaskOverviewIntent = new Intent(this, TaskListViewActivity.class);
-        startActivity(callTaskOverviewIntent);
-        this.finish();
+    private void showMessage(String message) {
+        Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_SHORT).show();
+                Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_INDEFINITE).show();
+    }
+
+    private void showPersistentMessage(String message) {
+        Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_INDEFINITE).show();
+    }
+
+
+    private void userDbInitialise() {
+        new Thread(() -> {
+            if (!(userDBOperation.authenticateUser("s@bht.de", "000000"))) {
+                userDBOperation.createUser(new User("s@bht.de","000000"));
+            }
+        }).start();
     }
```

Comment:
hier kann man die Methode für LoginState Bedingung schreiben

```suggestion
    private void handleUserLoginState(MainViewModel.LoginState loginState) {
            switch (loginState) {
                case INVALID_EMAIL:
                    showPersistentMessage(getString(R.string.login_username_error));
                    break;
                case WRONG_PASSWORD:
                    Snackbar.make(findViewById(R.id.rootView), getString(R.string.login_password_not_allowed), Snackbar.LENGTH_SHORT).show();
                    break;
                case AUTHENTICATION_FAIL:
                    showPersistentMessage(getString(R.string.login_authentication_failed));
                    break;
                case AUTHENTICATION_SUCCESS:
                    Intent callTaskListViewIntent = new Intent(this, TaskListViewActivity.class);
                    startActivity(callTaskListViewIntent);
                    this.finish();
                    break;
                default:
                    break;
            }
    }
```

Reply:
Sehr übersichtlich